test/services/database/comment/comment_repository_test.dart

```diff
@@ -0,0 +1,316 @@
+
+        final commentId = await commentRepo.addComment(post.id, commentData);
+
+        postComments.add(CommentFirestore(id: commentId, data: commentData));
+      }
+
+      // Check that the user comments have been added
+      for (user in users) {
+        final userComments = await commentRepo.getUserComments(user.uid);
+
+        expect(userComments, isNotEmpty);
+      }
+
+      return (users, postComments);
+    }
+
+    group("deleting comments", () {
+      test("should delete a comment", () async {
+        // Add a comment
+        final commentData =
+            postCommentDataGenerator.createMockCommentData(ownerId: user.uid);
+
+        final commentId = await commentRepo.addComment(post.id, commentData);
+
+        // Check that it was added correctly
+        await checkPostAndUserCommentsNotEmpty();
+
+        // Delete the comment
+        await commentRepo.deleteComment(post.id, commentId, user.uid);
+
+        // Check that it was deleted correctly
+        await checkPostAndUserCommentsEmpty();
+      });
+
+      test("should delete a comment when they are multiple comments", () async {
+        const nbUsers = 5;
+        late List<UserFirestore> users;
+        late List<CommentFirestore> postComments;
+
+        (users, postComments) = await addCommentsForUsers(nbUsers);
+
+        // Delete the comment modulo 2
+        for (final (i, user) in users.indexed) {
+          final comment = postComments[i];
+
+          if (i % 2 == 0) {
+            await commentRepo.deleteComment(post.id, comment.id, user.uid);
+          }
+        }
+
+        final actualPostComments = await commentRepo.getPostComments(post.id);
+
+        // Check that the right user and post comments were deleted
+        for (final (i, user) in users.indexed) {
+          final userComments = await commentRepo.getUserComments(user.uid);
+          final postComment = postComments[i];
+
+          if (i % 2 == 0) {
+            expect(userComments, isEmpty);
+            expect(actualPostComments.contains(postComment), isFalse);
+          } else {
+            expect(userComments, isNotEmpty);
+            expect(actualPostComments.contains(postComment), isTrue);
+          }
+        }
+      });
+    });
+
+    group("deleting all comments", () {
+      test("should delete all the comments", () async {
+        const nbUsers = 5;
+        late List<UserFirestore> users;
+
+        (users, _) = await addCommentsForUsers(nbUsers);
+
+        final batch = fakeFirestore.batch();
+        await commentRepo.deleteAllComments(post.id, batch);
+        await batch.commit();
+
+        // Check that all the comments were deleted
+        final actualPostComments = await commentRepo.getPostComments(post.id);
+        expect(actualPostComments, isEmpty);
+
+        for (user in users) {
+          final userComments = await commentRepo.getUserComments(user.uid);
+          expect(userComments, isEmpty);
+        }
+      });
```

Comment:
I think it is interesting to add a comment that is not deleted by the `deleteAll`:
```suggestion
      test("should delete all the comments", () async {
        const nbUsers = 5;
        late List<UserFirestore> users;

        // All users have a comment under the post
        (users, _) = await addCommentsForUsers(nbUsers);
        // The first user has a post in the other post
        final (_, expectedFirstUserComment) = await commentGenerator.addComment(
          otherPost.id,
          users.first.uid,
          commentRepo,
        );

        final firstUserComments = await commentRepo.getUserComments(
          users.first.uid,
        );
        expect(firstUserComments.length, equals(2));

        final batch = fakeFirestore.batch();
        await commentRepo.deleteAllComments(post.id, batch);
        await batch.commit();

        // Check that all the comments were deleted
        final actualPostComments = await commentRepo.getPostComments(post.id);
        expect(actualPostComments, isEmpty);

        // The first user is the only one with a comment left
        final actualFirstUserComments = await commentRepo.getUserComments(
          users.first.uid,
        );
        expect(actualFirstUserComments, equals([expectedFirstUserComment]));

        for (user in users.skip(1)) {
          final userComments = await commentRepo.getUserComments(user.uid);
          expect(userComments, isEmpty);
        }
      });
```

Reply:
Good idea !
Done in 310ffc73223c2f15a0af61690b660d8d6afe80df.